### DISCUSSION TITLE
EFI partition not defined

### DISCUSSION THREAD

**Discussion by fergrorke** [elementary/installer#866] [Q&A]
I'm trying to install Elementary OS 8 on my Windows 11 laptop. It was configured to dual boot with another Ubuntu-based distro which I want to overwrite. During the install process I can select the Linux (to be formatted), swap and boot/efi partitions but I get the following fail message:

ERROR: libdistinst: pointer to EFI is null
ERROR: libdistinst: pointer to EFI is null
ERROR: libdistinst: pointer to EFI is null
ERROR: errored while installing system: partition validation: EFI partition was not defined
INFO: Install error: partition validation: EFI partition was not defined

Any idea what going on here and how to overcome it?

**Comment by livid** [comment]
I have a similar setup and encountered the same error. I ended up installing Ubuntu 24.04, and its installer worked.

**Comment by fergrorke** [comment]
Yes, I gave up on the Elementary and installed KDE Plasma instead.

**Comment by autodafe665** [comment]
Same problem.

Checked me ssd with Gparted. It said there are not enough dosfstools and mtools.
After i`ve installed it, EFI partition become readable by elementary-os installer
But the problem hasn`t been solved.

**Comment by DoodMann** [comment]
I know what's the problem, or at least I think I do.

Windows allocates only ~100MB worth of disk space for the EFI partition. This somehow doesn't play well with EOS 8. However, when I tried wiping the whole SSD and made a 500MB EFI partition, it recognized it. I think it's some kind of storage limitation (though Fedora and Zorin can install to it no problem :P)

**Reply by ArskaBoii** [reply]
So I have the same issue with the Elementary OS installer. Should I delete my current EFI partition when I have Windows 10 already installed on my new SSD and then make a new EFI partition that is 500 MB in size?

**Reply by vjr** [reply]
> So I have the same issue with the Elementary OS installer. Should I delete my current EFI partition when I have Windows 10 already installed on my new SSD and then make a new EFI partition that is 500 MB in size?

Do not delete your current EFI partition which is required by Windows - just create a new EFI partition of some good size like 600 MB and mount at /boot/efi then proceed with your installation?

**Comment by SamuGallo-06** [comment]
In my case, even creating a 1Gb EFI partition, the installer warns me that the` EFI partition is too small` during the manual partitioning. 

In a 1Tb nvme ssd, i've created 4 partitions:
- EFI: 512Mb
- Swap: 32 Gb
- Root (/): 128 Gb
- /home: remaning space

Using the "Erase and install" option, it shows the error `EFI partition was not defined`
I'm installing EOS 8 on an iMac 18,3

**Comment by GwynethLlewelyn** [comment]
I've encountered the same, or a very similar, issue.

There are two disks on the PC I wish to double-boot Windows and elementary OS from: the first, the older one, just has an MBR, the rest being mostly (unbootable) backups. One (extended) partition of 256 GB has been set apart for elementary OS (`/boot/efi` and `/`).

The second disk is a Samsung SSD. It is also an MBR drive; however, it features a 0.5 TB EFI partition, followed by the actual (bootable) Windows 10 partition. This setup is a weird legacy which I never bothered to 'correct', since that almost always leads to data loss.

My intention is to have the SSD's first partition to become a true EFI partition, have the system boot from it, and then have the choice of either running Windows (from the same SSD disk) or elementary OS from the old disk instead.

However, the elementary OS installer chokes on this configuration with the errors shown by the OP.

**Comment by thallesgate** [comment]
Exactly the same issue here. I had to create another 512MB EFI partition and select that one.
